### ISSUE TITLE
No way to remove GPX trace marker

### ISSUE THREAD

**Issue by hikemaniac** [openstreetmap/openstreetmap-website#1735]
When you select a GPX trace and press "map", the OpenStreetMap map view opens and a blue marker appears, showing the starting point of the trace. Unfortunately there is no way to remove the marker from the map again, as it is URL set. Neither clicking it nor moving the map somewhere else helps. The only way is to open osm.org in a new tab. Can someone please consider adding a removing option, because the marker is really annoying in some cases.
cc @biswesh456

**Comment by tomhughes** [comment]
I can't think of any obvious way to provide this, so my preference would be to back out the commit that adds the marker - it is only of marginal benefit.

**Comment by biswesh456** [comment]
We can provide a link to remove the marker which would actually redirect to the url without the latitudes and longitudes.

**Comment by tomhughes** [comment]
Well we can, but there isn't really a good place to put it. If it opened the side panel to show the trace details or something then you could have it remove the marker when the side panel was closed but that's a rather bigger project ;-)

**Comment by hikemaniac** [comment]
How about opening a small popup?

![gpx marker proposed](https://user-images.githubusercontent.com/31667811/36074621-04868e96-0f43-11e8-84af-6f8981117ab2.gif)

**Comment by mmd-osm** [comment]
You could also manually remove "`mlat=...&mlon=...`" from the URL to get rid of the trace marker.

**Comment by danieldegroot2** [comment]
Was #670

@AntonKhorev Would it be possible to open a single item menu on left and/or right click with 'remove marker' (and the right click menu)? (if user uses (right) click at/near marker user probably wants to click on marker)
A marker is a (only slightly intrusive) 'pop-up'(/overlay/layer). The user should be able to close such elements easily.

Sidenote: Currently the cursor on hover is a 'click'/url type cursor, but nothing happens on (left) click. If solution is not feasible, this cursor icon might be changed to a 'pointer' type cursor to indicate there is no interaction.

The user can already
- remove the marker if they were sharing the link, by closing the share tab.
- remove a note marker when the note layer is enabled by disabling the note layer in the layer menu.
- remove a route marker by closing the route tab (which will keep only the end point / location search)
- remove a location search by closing the search tab
(if user wants to remove search while typing then they have to manually (or) Ctrl+A+Delete remove it, there is no ❌ small cross icon remove button next to search; there might've been a ticket for this at some point.)

Related #737 (which is its own page, would require different solution if any)

**Comment by AntonKhorev** [comment]
> The user can already
> 
>    * (a list of completely different things)

...and the solutions proposed here are *let's do another completely different thing like a popup menu*.

If we don't want to introduce *another completely different thing*, one possible solution is: make the `mlat`/`mlon` marker to be managed by the *Share* panel. Currently you can get two visible markers at the same time: one from `mlat`/`mlon` and one from opening *Share*. This is also bad probably. Instead we could:

- have one marker managed by *Share* panel code
- that marker will appear when the map is opened with the `mlat`/`mlon` parameters
- that marker won't appear/disappear when the share panel is opened/closed, otherwise sharing a marker won't work as expected
- maybe clicking the marker should open the *Share* panel if it was closed
- the visibility of the marker is controlled by the *Include marker* checkbox; unchecking it is going to be a way to remove the marker
- maybe the checkbox needs to be replaced by something more prominent, like a button [Add Marker] / [Remove Marker]; then clicking the marker will open the *Share* panel and focus on the [Remove Marker] button

**Comment by danieldegroot2** [comment]
Yes, that would be my other suggestion, but is more noticeable change / might be more work to redo rather than add code, though I suppose linking pop-up to marker is the same thing and does not solve below directly.

> Currently you can get two visible markers at the same time: one from mlat/mlon and one from opening Share. This is also bad probably.

Yes, I noticed as well. I agree it is confusing/inconvenient(, but user could manually check it, right..).
Note, the coordinate parameter is still for the old marker. 'Add marker' does not add coordinate parameter when creating new marker. Maybe change(/clear) the url parameter when moving marker?
(sidenote: the behaviour for 'Add note' marker can be kept as-is)
